### PR TITLE
feat: Add SyncResource CRD

### PR DESCRIPTION
**Pull request by ruanxin**

### CODE REVIEW

File: api/v1alpha1/syncresource_types.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+package v1alpha1
+
+import (
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (nesmabadr) [inline api/v1alpha1/syncresource_types.go:26]: Please remove these lines

File: config/rbac/syncresource_editor_role.yaml
Comment (nesmabadr) [inline config/rbac/syncresource_editor_role.yaml:1]: Should these rbac files be referenced anywhere?

Comment (ruanxin) [reply config/rbac/syncresource_editor_role.yaml:1]: They are automatically generated by kubebuilder, basically kubebuilder offers two ways to config `ClusterRole`, reference resource role (by `xxx_editor_role`, `xxx_viewer_role`) specifically or use `role.yaml` https://github.com/kyma-project/lifecycle-manager/blob/e47431db6cedd2b74d4fd66cff40f6468e2766b9/config/rbac/kustomization.yaml#L11 which include all resources, let's have another issue align about which way prefer, then do the cleanup.

Comment (ruanxin) [reply config/rbac/syncresource_editor_role.yaml:1]: I changed my mind, we need to only keep `role.yaml`, there are problems in those editor and viewer role, check the followup [issue](https://github.com/kyma-project/lifecycle-manager/issues/1424) for detail.

**Pull request merged**